frontend/src/scenes/notebooks/Nodes/NodeWrapper.tsx

```diff
@@ -115,6 +116,14 @@ export function NodeWrapper<T extends CustomNotebookNodeAttributes>({
     }, [resizeable, updateAttributes])
 
     const parsedHref = typeof href === 'function' ? href(attributes) : href
+    // If a title is set on the attrs we use it. Otherwise we use the base component title.
+    const title = useMemo(() => {
+        return attributes.title
+            ? attributes.title
+            : typeof defaultTitle === 'function'
+            ? defaultTitle(attributes)
```

Comment:
It looks like `defaultTitle` is only ever a string now. Maybe we should limit the types to `string`

Reply:
Good point!